### ISSUE TITLE
Listen receiving ZoneAwarePromise instead of a function

### ISSUE THREAD

**Issue by MarcoBomfim** [developit/mitt#113]
Hey there, 

I'm using mitt in a project that is currently a micro front-end platform. In order for applications of different frameworks to listen and react to changes from one another, I've created a service, which exports both listen and dispatch functions from mitt. This is a rough sketch of it:
```
// EventBus here is the mitt() function
export default function ({ config, eventBus }) {
  const dispatch = ({ name }) => {
    return eventBus.emit(name)
  }

  const listen = ({ name, callback = () => {} }) => {
    return eventBus.on(name)
  }

  return {
    dispatch,
    listen,
  }
}
```

This works fine for any app using React or Vue, and even some vanilla JS services. The issue here is that I have an Angular application, which whenever I call `eventBus.listen({ name: 'someEvent', callback: () => this.somecallback() })`, when a component is mounted, it runs fine, but when an event is triggered by some other application, and this listeners hears the `someEvent`, it gives me the following error:
![image](https://user-images.githubusercontent.com/30595009/88575163-ce2c7000-d019-11ea-94fe-3eef1e08f613.png)

This might probably be a very stupid mistake from my part, but I can't seem to find a way to use this service along with this mitt() interface I've built. Is there a way to send promises do the `on()` method from mitt?

**Comment by developit** [comment]
Are you sure this isn't just because some code is returning a native Promise to Angular? Angular relies on a patched version of Promise in order to handle change detection.

**Comment by MarcoBomfim** [comment]
This happens whether I use a simple console.log(), or an empty function as well, so I'm not sure what could be turning the function into a promise. 

Offtopic question: If I create a new instance of Mitt in my Angular project, would it be able to listen events from another instance?

**Comment by MarcoBomfim** [comment]
Alright, I've decided to create two new methods specific for Angular, which unfortunately uses `document.dispatchEvent` and `document.addEventListener`. After tweaking with the code a little bit, I've managed to assign the return of the promise to an instance variable inside the component, and then, called out the event listener.

The problem is that for whatever reason, Angular runs the method that calls mitt's `on`, but it runs only **once**, if you trigger an event from elsewhere, and wait for the listener to execute the callback, is as it has never existed. The only way I managed to make it work was by using the document itsel. It's not pretty, but this Angular stack should be gone in the future, so whatever. 

Thanks @developit for the attention. Great work with the project!
